internal/server/ranking.go

```diff
@@ -69,6 +77,31 @@ func (a SeriesByRank) Less(i, j int) bool {
 	if scorei != scorej {
 		return scorei < scorej
 	}
+
+	datesi := []string{}
+	for date := range a[i].Val {
+		datesi = append(datesi, date)
+	}
+	sort.Strings(datesi)
+	latesti := datesi[len(datesi)-1]
```

Comment:
If you're looking anyway, it might be more efficient to compute max while you do that, rather than do another Sort?

Reply:
Done